pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -142,16 +142,25 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return common.ResultRequeue, err
 		}
 		subnetPort.Status.Attachment = v1alpha1.PortAttachment{ID: *nsxSubnetPortState.Attachment.Id}
-		subnetPort.Status.NetworkInterfaceConfig = v1alpha1.NetworkInterfaceConfig{
-			IPAddresses: []v1alpha1.NetworkInterfaceIPAddress{
-				{
-					Gateway: "",
-				},
-			},
-		}
+		subnetPort.Status.NetworkInterfaceConfig = v1alpha1.NetworkInterfaceConfig{}
 		if len(nsxSubnetPortState.RealizedBindings) > 0 {
-			subnetPort.Status.NetworkInterfaceConfig.IPAddresses[0].IPAddress = *nsxSubnetPortState.RealizedBindings[0].Binding.IpAddress
+			for _, rb := range nsxSubnetPortState.RealizedBindings {
+				subnetPort.Status.NetworkInterfaceConfig.IPAddresses = append(
+					subnetPort.Status.NetworkInterfaceConfig.IPAddresses,
+					v1alpha1.NetworkInterfaceIPAddress{
+						IPAddress: *rb.Binding.IpAddress,
+					},
+				)
+			}
 			subnetPort.Status.NetworkInterfaceConfig.MACAddress = strings.Trim(*nsxSubnetPortState.RealizedBindings[0].Binding.MacAddress, "\"")
```

Comment:
I do not get this. Do you mean in realized_bindings, there might be multiple IP addresses, but we only get the first mac address? 

I found below from NSX API doc.  

"realized_bindings": [
    {
      "source": "VM_TOOLS",
      "binding": {
        "ip_address": "192.168.100.162",
        "mac_address": "00:0c:29:46:b0:72",
        "vlan": 0
      },
      "binding_timestamp": 1591432722000
    },
    {
      "source": "ARP_SNOOPING",
      "binding": {
        "ip_address": "192.168.100.162",
        "mac_address": "00:0c:29:46:b0:72",
        "vlan": 0
      },
      "binding_timestamp": 1591434843090
    }
  ],

Reply:
Based on the discussion offline, I changed this back to only taking the first one from the realizedBindings/addressBindings as NSX only supports one IP per SubnetPort. We can revisit this if NSX vpc supports ipv6 in the future.